File: jenkins-plugin/src/test/java/org/eazyportal/plugin/release/jenkins/action/SetSnapshotVersionActionFactoryTest.java
package org.eazyportal.plugin.release.jenkins.action;

import org.eazyportal.plugin.release.core.action.SetSnapshotVersionAction;
import org.eazyportal.plugin.release.core.scm.ScmActions;
import org.eazyportal.plugin.release.core.scm.model.ScmConfig;
import org.eazyportal.plugin.release.core.version.SnapshotVersionProvider;
import org.eazyportal.plugin.release.jenkins.ReleaseStepConfig;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.verifyNoInteractions;
import static org.mockito.Mockito.verifyNoMoreInteractions;
import static org.mockito.Mockito.when;

class SetSnapshotVersionActionFactoryTest {

    @Mock
    private transient ReleaseStepConfig releaseStepConfig;
    @Mock
    private transient SnapshotVersionProvider snapshotVersionProvider;

    @InjectMocks
    private SetSnapshotVersionActionFactory underTest;

    @BeforeEach
    void setUp() {
        MockitoAnnotations.openMocks(this);
    }

    @Test
    void test_create() {
        // GIVEN
        ScmActions scmActions = mock(ScmActions.class);

        // WHEN
        when(releaseStepConfig.getScmConfig()).thenReturn(ScmConfig.getGIT_FLOW());

        // THEN
        SetSnapshotVersionAction actual = underTest.create(scmActions);

        assertThat(actual).hasNoNullFieldsOrProperties();

        verifyNoInteractions(scmActions, snapshotVersionProvider);
        verify(releaseStepConfig).getScmConfig();
        verifyNoMoreInteractions(releaseStepConfig);
    }

}

File: gradle-plugin/src/test/kotlin/org/eazyportal/plugin/release/gradle/action/SetSnapshotVersionActionFactoryTest.kt
package org.eazyportal.plugin.release.gradle.action

import org.assertj.core.api.Assertions
import org.eazyportal.plugin.release.core.project.ProjectActionsFactory
import org.eazyportal.plugin.release.core.scm.ScmActions
import org.eazyportal.plugin.release.core.scm.model.ScmConfig
import org.eazyportal.plugin.release.gradle.model.EazyReleasePluginExtension
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.mockito.kotlin.mock
import org.mockito.kotlin.whenever

internal class SetSnapshotVersionActionFactoryTest {

    private lateinit var underTest: SetSnapshotVersionActionFactory

    @BeforeEach
    fun setUp() {
        underTest = SetSnapshotVersionActionFactory()
    }

    @Test
    fun test_create() {
        val extension: EazyReleasePluginExtension = mock()

        val projectActionsFactory: ProjectActionsFactory = mock()
        val scmActions: ScmActions = mock()
        val scmConfig: ScmConfig = mock()

        // WHEN
        whenever(extension.projectActionsFactory).thenReturn(projectActionsFactory)
        whenever(extension.scmActions).thenReturn(scmActions)
        whenever(extension.scmConfig).thenReturn(scmConfig)

        // THEN
        val actual = underTest.create(extension)

        Assertions.assertThat(actual).hasNoNullFieldsOrProperties()
    }

}
